config/defaults_gen/defaults_gen.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/algorand/go-algorand/config"
+)
+
+const defaultLocalVariableDeclaration = "var defaultLocal"
+
+var outputfilename = flag.String("o", "", "Name of the file where the generated datastructure would go to.")
+var packageName = flag.String("p", "", "Name of the package.")
+var headerFileName = flag.String("h", "", "Name of the header filename")
+var jsonExampleFileName = flag.String("j", "", "Name of the json example file")
+
+var autoGenHeader = `
+// This file was auto generated by ./config/defaults_gen/defaults_gen.go, and SHOULD NOT BE MODIFIED in any way
+// If you want to make changes to this file, make the corresponding changes to Local in config.go and run "go generate".
+`
+
+func main() {
+
+	flag.Parse()
+	if *outputfilename == "" || *packageName == "" || *headerFileName == "" || *jsonExampleFileName == "" {
+		fmt.Printf("one or more of the required input arguments was not provided\n")
```

Comment:
(Repeating an earlier comment) I think adding `os.Args[0]` to the error string is good because if code generation fails, we'll be able to trace the failure to this tool or rule it out as cause of failure.